Frameworks/CoreData/NSAttributeDescription.mm

```diff
@@ -14,8 +14,111 @@
+            return valueString;
+        case NSBooleanAttributeType:
+            return [NSNumber numberWithBool:[valueString boolValue]];
+        default:
+            // This type can't have a parsed default.
+            return nil;
+    }
+    return nil;
+}
+
+- (id)initWithXMLElementName:(NSString*)entityName attributes:(NSDictionary<NSString*, NSString*>*)attributes {
+    if (self = [super initWithXMLElementName:entityName attributes:attributes]) {
+        _attributeType = [[self class] _attributeTypeFromString:attributes[@"attributeType"]];
+        _valueTransformerName = [attributes[@"valueTransformerName"] copy];
+
+        if (_attributeType == NSDateAttributeType) {
+            NSString* defaultDateString = attributes[@"defaultDateTimeInterval"];
+            NSString* minDateString = attributes[@"minDateTimeInterval"];
+            NSString* maxDateString = attributes[@"maxDateTimeInterval"];
+
+            _defaultValue = [[NSDate alloc] initWithTimeIntervalSinceReferenceDate:[defaultDateString longLongValue]];
+            _minValue = [NSDate dateWithTimeIntervalSinceReferenceDate:[minDateString longLongValue]];
+            _maxValue = [NSDate dateWithTimeIntervalSinceReferenceDate:[maxDateString longLongValue]];
+        } else {
+            _defaultValue = [[[self class] _convertValueString:attributes[@"defaultValueString"] forAttributeType:_attributeType] copy];
+            _minValue = [[self class] _convertValueString:attributes[@"minValueString"] forAttributeType:_attributeType];
+            _maxValue = [[self class] _convertValueString:attributes[@"maxValueString"] forAttributeType:_attributeType];
+        }
+
+        NSMutableArray<NSPredicate*>* validationPredicates = [NSMutableArray array];
+        if (_minValue) {
+            [validationPredicates addObject:[NSPredicate predicateWithFormat:@"SELF >= %@", (id)_minValue]];
+        }
+
+        if (_maxValue) {
+            [validationPredicates addObject:[NSPredicate predicateWithFormat:@"SELF <= %@", (id)_maxValue]];
+        }
+
+        if (validationPredicates.count > 0) {
+            self.validationPredicates = validationPredicates;
+        }
+    }
+    return self;
+}
+
+- (void)dealloc {
+    [_defaultValue release];
+    [_valueTransformerName release];
+    [super dealloc];
+}
+
+- (NSString*)description {
+    return [NSString stringWithFormat:@"<%@ %p: %@,%@%@%@%@%@ default=%@ userInfo=%@>",
+                                      object_getClass(self),
+                                      self,
+                                      self.name,
+                                      self.optional ? @" optional" : @"",
+                                      self.transient ? @" transient" : @"",
+                                      self.indexed ? @" indexed" : @"",
+                                      self.storedInExternalRecord ? @" external" : @"",
+                                      self.indexedBySpotlight ? @" spotlight" : @"",
```

Comment:
Likely, but this is not intended to be used supremely often. Logging Entity, attribute, and property descriptions is not a use case that needs to be performant.